### PR TITLE
Test API modules in pyface.data_view

### PR DESCRIPTION
**Pull request by kitchoi**
Targeting #750

This PR
- Adds tests for the api modules.
- Allows NumPy to be absent and still be able to import data models that don't require NumPy

The treatment mirrors the import of ArrayEditor in TraitsUI:
https://github.com/enthought/traitsui/blob/7b53731606959dfaa7b2e3e8089d00e7a771e312/traitsui/api.py#L29-L33

Ideally we should have a CI job that has the most minimal requirements, and in the case of Qt, NumPy is not an install requirement. I just ran the tests locally with NumPy removed from my environment.

### CODE REVIEW

File: pyface/data_view/tests/test_api.py
Comment (kitchoi) [reply pyface/data_view/tests/test_api.py:17]: Ummmm I could try to group them by the modules they are coming from, but that's quite laborious.
If I try to group them by their actual meaning/usage, how I categorize things may be incorrect, and it also means the next developer will have to think about where to add their new import.

For your concern, I'd propose trying to count and assert the number of items in the api, but at this level the marginal benefit is getting smaller.

Comment (kitchoi) [reply pyface/data_view/tests/test_api.py:17]: > For your concern, I'd propose trying to count and assert the number of items in the api, but at this level the marginal benefit is getting smaller.

Done.

**Review by rahulporuri — APPROVED** [review]
LGTM.

**Comment by rahulporuri** [comment]
@kitchoi i'm not sure if you want to wait on @corranwebster s review for this PR.

**Comment by kitchoi** [comment]
Ah I noticed that #750 has now changed such that the tests here would fail. @corranwebster I'd leave this to you as to what to do with this PR. Feel free to close it / cherry-pick changes from here instead.

**Comment by corranwebster** [comment]
Looks OK, not a fan of all the ad-hoc numpy import testing; I'd really like that in just one place in all of ETS, but that's orthogonal to this PR.

I think counting imports is just going to be annoying - the API tests should just be smoke tests to catch dumb errors like a bad import.  For example, the tests are currently broken because I just added a new import to the API module in the branch this being merged into.

**Comment by corranwebster** [comment]
Probably easies just to merge and fix in the other branch.

**Comment by kitchoi** [comment]
> I think counting imports is just going to be annoying - the API tests should just be smoke tests to catch dumb errors like a bad import. For example, the tests are currently broken because I just added a new import to the API module in the branch this being merged into.

I think I am in mind on this. Yes one needs to modify the test when modifying the api module, but the api module is not changed all that often. I think I still prefer keeping the api test current because this forces the developer to consider the dependency implications when they add an item to the api module. This is done in some other projects too, so this is not the first.

**Comment by kitchoi** [comment]
> I think I am in mind on this.

Typo: I think I am in two minds on this

**Pull request merged**
